build.sbt

```diff
@@ -25,7 +25,9 @@ scalacOptions ++= Seq(
   "-Wvalue-discard",            // Warn when non-Unit expression results are unused.
   "-Ywarn-macros:after",        // Needed for correct implicit resolution.
   "-Wconf:cat=unused-nowarn:s", // Silence nowarn usage warnings.
-  "-Xfatal-warnings"            // Fail the compilation if there are any warnings.
+  "-Xfatal-warnings",            // Fail the compilation if there are any warnings.
+  "-Ymacro-annotations",         // Bring textual abstraction to the level of definitions.
+  "-Wconf:src=*<macro>*&cat=unused:s" // zio-mock works with it
```

Comment:
Do we really need it? 

Could you please share a link where this instruction is described?

Reply:
So, I've looked into it and yes, we have to silence these warnings

I suggest adding this option separately  for `root` project to don't mix common options with specific ones. Also, I suggest enabling this option only for test scope. After a little refactoring of the `root` project, it might look like this:
```scala
lazy val root = project
  ...
  .settings(
    libraryDependencies += zio,
  ...
  .settings(
    libraryDependencies ++= zioTest,
    libraryDependencies += zioMock,
    Test / scalacOptions += "-Wconf:src=*<macro>*&cat=unused:s" // Silence warns of unused things for macro in zio mock
  )
```

Reply:
Unfortunately it doesn't work for test scope, I just separated it via `.settings`

Reply:
Hm, it worked on my local machine

Can you try clean, compile, and then run tests again?